extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/remote_inference_client/text_2_speech.py

```diff
@@ -0,0 +1,309 @@
+import base64
+import copy
+import io
+import json
+import numpy as np
+from typing import TYPE_CHECKING, Any, List, Optional
+
+# HuggingFace API imports
+from huggingface_hub import InferenceClient
+
+from aiconfig import CallbackEvent
+from aiconfig.default_parsers.parameterized_model_parser import (
+    ParameterizedModelParser,
+)
+from aiconfig.model_parser import InferenceOptions
+from aiconfig.schema import (
+    ExecuteResult,
+    Output,
+    OutputDataWithStringValue,
+    Prompt,
+    PromptMetadata,
+)
+from aiconfig.util.config_utils import get_api_key_from_environment
+from aiconfig.util.params import resolve_prompt
+from scipy.io.wavfile import write as write_wav
+
+
+# Circuluar Dependency Type Hints
+if TYPE_CHECKING:
+    from aiconfig.Config import AIConfigRuntime
+
+
+# Step 1: define Helpers
+def refine_completion_params(model_settings: dict[Any, Any]) -> dict[str, Any]:
+    """
+    Refines the completion params for the HF text_to_speech api. Removes any unsupported params.
+    See https://github.com/huggingface/huggingface_hub/blob/main/src/huggingface_hub/inference/_client.py#L1624
+    for supported params.
+    """
+
+    supported_keys = {
+        "model",
+    }
+
+    completion_data = {}
+    for key in model_settings:
+        if key.lower() in supported_keys:
+            completion_data[key.lower()] = model_settings[key]
+
+    return completion_data
+
+
+def construct_output(audio: bytes) -> Output:
+    def _b64_encode_bytes(byte_array: bytes) -> str:
+        return base64.b64encode(byte_array).decode("utf-8")
+
+    data = dict(
+        kind="base64",
+        value=_b64_encode_bytes(audio),
+    )
+
+    # TODO: Do we need to encode this for different mime types? For now, we
+    # just set to audio/wav (works for a few models tested)
```

Comment:
took a look around but couldn't find any documentation for this. They use python's Ipython (jupyter) audio module which is mainly for wav, so +1 i think this should be fine for now.